tensorboard/util.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""TensorBoard helper routine module.
+
+This module is a trove of succinct generic helper routines that don't
+pull in any heavyweight dependencies aside from TensorFlow.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import locale
+
+
+def closeable(klass):
+  """Makes a class with a close method able to be a context manager.
+
+  This decorator is a great way to avoid having to choose between the
+  boilerplate of __enter__ and __exit__ methods, versus the boilerplate
+  of using contextlib.closing on every with statement.
+
+  Args:
+    klass: The class being decorated.
+
+  Raises:
+    ValueError: If class didn't have a close method, or already
+        implements __enter__ or __exit__.
+  """
+  if 'close' not in klass.__dict__:
+    # coffee is for closers
+    raise ValueError('Class does not define a close() method: ' + klass)
```

Comment:
`TypeError: cannot concatenate 'str' and 'type' objects` (prefer `repr(klass)` or `klass.__name__`)